src/crimson/osd/osd_operation.h

```diff
@@ -269,14 +282,32 @@ class OperationRegistry {
     static_cast<int>(OperationTypeCode::last_op)
   > op_id_counters = {};
 
+  seastar::timer<seastar::lowres_clock> shutdown_timer;
+  seastar::promise<> shutdown;
 public:
+  OperationRegistry()
+    : shutdown_timer([this] {
+	if (std::all_of(registries.begin(),
```

Comment:
can we set the callback only when doing shutdown? i think we don't need to set the call back until needed for smaller memory footprint, as presumably an empty `noncopyable_function` use smaller memory than the one holding a function. and for better readability.